### PR TITLE
fix(kit): `InputFlies` new value of i18n-token `TUI_INPUT_FILE_TEXTS` triggers text changes

### LINKED ISSUE #2709: 🐞 - `InputFlies` no text changes with new value of `TUI_INPUT_FILE_TEXTS` (i18n-token)
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://stackblitz.com/edit/angular-xncuqv

### Is this issue blocking you?

Blocking

### Description

При переопределении значения токена `TUI_INPUT_FILE_TEXTS ` на поток, который при смене локализации выдает новые значения текстов, значения `maxSizeRejectionReason` или `formatRejectionReason` в уже отрисованном rejectedFile не меняются на новые при динамической смене локализации

### Angular version

12.2.0

### Taiga UI version

2.62.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [X] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [X] Windows
- [ ] Linux
- [X] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2709

## What is the new behavior?
You can dynamically change language of error messages for rejected files.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            hideLanguageSwitcher: false,
+        });
+
+        cy.get(`#demoContent input[type="file"]`).selectFile(
+            `cypress/fixtures/stubs/web-api.svg`,
+        );
+
+        cy.get(`#demoContent tui-file`)
+            .should(`contain.text`, `Wrong file type`)
+            .matchImageSnapshot(`01-01-input-files-[english]-wrong-file-type`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:16]: 1-01-01-input-files- english -wrong-file-type snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+        cy.tuiSetLanguage(`dutch`);
+
+        cy.get(`#demoContent tui-file`)
+            .should(`contain.text`, `Verkeerd bestandsformaat`)
+            .matchImageSnapshot(`01-02-input-files-[dutch]-wrong-file-type`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:22]: 1-01-02-input-files- dutch -wrong-file-type snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            hideLanguageSwitcher: false,
+        });
+
+        cy.get(`#demoContent input[type="file"]`).selectFile(
+            `cypress/fixtures/stubs/web-api.svg`,
+        );
+
+        cy.get(`#demoContent tui-file`)
+            .should(`contain.text`, `File is too large 2 KB`)
+            .matchImageSnapshot(`02-01-input-files-[english]-file-too-large`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:39]: 1-02-01-input-files- english -file-too-large snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        cy.tuiSetLanguage(`dutch`);
+
+        cy.get(`#demoContent tui-file`)
+            .should(`contain.text`, `Bestandsgrootte overschreden 2 KB`)
+            .matchImageSnapshot(`02-02-input-files-[dutch]-file-too-large`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:45]: 1-02-02-input-files- dutch -file-too-large snap

**Review by splincode — APPROVED** [review]

File: projects/kit/components/input-files/input-files.component.ts [hunk 5]
```diff
@@ -140,25 +140,33 @@ export class TuiInputFilesComponent
...
 
+    @tuiPure
+    getMaxSizeRejectionError$(maxFileSize: number): Observable<string> {
+        return combineLatest([this.inputFileTexts$, this.units$]).pipe(
+            map(
+                ([{maxSizeRejectionReason}, units]) =>
+                    maxSizeRejectionReason + tuiFormatSize(units, maxFileSize),
+            ),
+        );
+    }
```
Comment (splincode) [inline projects/kit/components/input-files/input-files.component.ts:151]: can make it better in a separate pipe?

```
<ng-template #maxSizeRejection>
    {{ maxFileSize | maxSizeRejectionError | async ?? '' }}
</ng-template>
```

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                hideLanguageSwitcher: false,
+            });
+
+            cy.get(`#demoContent input[type="file"]`).selectFile(
+                `cypress/fixtures/stubs/web-api.svg`,
+            );
+
+            cy.get(`#demoContent tui-file`)
+                .should(`contain.text`, `Wrong file type`)
+                .matchImageSnapshot(`01-01-input-files-[english]-wrong-file-type`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:17]: 1-01-01-input-files- english -wrong-file-type snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+            cy.tuiSetLanguage(`dutch`);
+
+            cy.get(`#demoContent tui-file`)
+                .should(`contain.text`, `Verkeerd bestandsformaat`)
+                .matchImageSnapshot(`01-02-input-files-[dutch]-wrong-file-type`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:23]: 1-01-02-input-files- dutch -wrong-file-type snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                hideLanguageSwitcher: false,
+            });
+
+            cy.get(`#demoContent input[type="file"]`).selectFile(
+                `cypress/fixtures/stubs/web-api.svg`,
+            );
+
+            cy.get(`#demoContent tui-file`)
+                .should(`contain.text`, `File is too large 2 KB`)
+                .matchImageSnapshot(`02-01-input-files-[english]-file-too-large`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:40]: 1-02-01-input-files- english -file-too-large snap

File: projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+            cy.tuiSetLanguage(`dutch`);
+
+            cy.get(`#demoContent tui-file`)
+                .should(`contain.text`, `Bestandsgrootte overschreden 2 KB`)
+                .matchImageSnapshot(`02-02-input-files-[dutch]-file-too-large`);
```
Comment (nsbarsukov) [inline projects/demo-integrations/cypress/tests/kit/input-files/input-files.spec.ts:46]: 1-02-02-input-files- dutch -file-too-large snap

File: projects/kit/components/input-files/input-files.component.ts (hunk 5 above)
Comment (nsbarsukov) [reply projects/kit/components/input-files/input-files.component.ts]: done ✅

**Pull request merged**